drivers/spi/spi_ifx_cat1.c

```diff
@@ -0,0 +1,356 @@
+/*
+ * Copyright 2023 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT infineon_cat1_spi
+
+#define LOG_LEVEL CONFIG_SPI_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(cat1_spi);
+
+#include "spi_context.h"
+
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/kernel.h>
+
+#include <cyhal_scb_common.h>
+#include <cyhal_spi.h>
+
+#define IFX_CAT1_SPI_LOCK_TMOUT_MS      (30 * 1000)
+#define IFX_CAT1_SPI_DEFAULT_OVERSAMPLE (4)
+#define IFX_CAT1_SPI_MIN_DATA_WIDTH     (8)
+#define IFX_CAT1_SPI_MAX_DATA_WIDTH     (32)
+
+/* Device config structure */
+struct ifx_cat1_spi_config {
+	CySCB_Type *reg_addr;
+	const struct pinctrl_dev_config *pcfg;
+	cy_stc_scb_spi_config_t scb_spi_config;
+	uint8_t irq_priority;
+};
+
+/* Data structure */
+struct ifx_cat1_spi_data {
+	struct spi_context ctx;
+	cyhal_spi_t obj; /* SPI CYHAL object */
+	cyhal_resource_inst_t hw_resource;
+	struct spi_config spi_cfg_cache;
+	size_t chunk_len;
+};
+
+static int32_t _get_hw_block_num(CySCB_Type *reg_addr)
+{
+	uint32_t i;
+
+	for (i = 0u; i < _SCB_ARRAY_SIZE; i++) {
+		if (_CYHAL_SCB_BASE_ADDRESSES[i] == reg_addr) {
+			return i;
+		}
+	}
+
+	return -ENOMEM;
+}
+
+static uint8_t _get_dfs_value(struct spi_context *ctx)
```

Comment:
Instead of spi_cfg_cache (see comment below), the dfs value could be stored. Running a function all the time for such important value is perhaps too heavy.